src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1131,6 +1131,15 @@ public static bool IsParameterTypeContext(this SyntaxTree syntaxTree, int positi
             return false;
         }
 
+        public static bool IsPossibleTupleElementNameContext(
+            this SyntaxTree syntaxTree, int position, SyntaxToken tokenOnLeftOfPosition, CancellationToken cancellationToken)
+        {
+            var token = tokenOnLeftOfPosition;
+            token = token.GetPreviousTokenIfTouchingWord(position);
+
+            return token.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken);
```

Comment:
If you type `(1`, there is no tuple literal in the syntax tree yet. 
Once you reach the first comma, then we're good though.

So I'm thinking to check whether the parent is an invocation, rather than check if it's a tuple literal. Then `Foo(X` will work properly.